pkg/api/handlers.go

```diff
@@ -39,6 +43,37 @@ func newHandler(context *clusterd.Context, config *Config) *Handler {
 	}
 }
 
+// RegisterMetrics registers all collected metrics by this API server.  Note this should be called in a
+// goroutine because it will retry upon failure and block until successful.
+func (h *Handler) RegisterMetrics() error {
```

Comment:
We always want to at least try to register with Prometheus because even if it's not currently running, you can never be sure if a Prometheus server will come up in the environment later on.  Note that registering with Prometheus is actually a local operation and does not require a Prometheus server to be running at registration time.  If one were to come up later on, we don't need to re-register again.

Something else to note here is that the way the CephExporter was written (from digitalocean's repo), it assumes a ceph connection is available at construction time.  Instead of code churn and refactoring on that stable codebase, I'm OK with keeping 1 ceph connection around even if it isn't eventually used by Prometheus scraping our metrics.